sdk/iot/core/doc/mqtt_state_machine.md

```diff
@@ -136,3 +138,120 @@ _Example:_
+
+#### Error policy
+
+The SDK will not handle protocol-level (WebSocket, MQTT, TLS or TCP) errors. The application-developer is expected to classify and handle errors the following way:
+
+- Authentication errors should not be retried.
+- Communication-related errors other than ones security-related should be considered retriable.
+
+Both IoT Hub and Provisioning services will use `MQTT CONNACK` as described in Section 3.2.2.3 of the [MQTT v3 specification](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Table_3.1_-).
+
+##### IoT Hub Service Errors
+
+APIs using `az_iot_hub_client_status` report service-side errors to the client through the IoT Hub protocols. (At the time of writing, only Twin responses may return errors.)
+
+The following APIs may be used to determine if the status indicates an error and if the operation should be retried:
+
+```C
+az_iot_hub_client_status status = response.status;
+if (az_iot_hub_client_is_success_status(status))
+{
+    // success case 
+}
+else
+{
+    if (az_iot_hub_client_is_retriable_status(status))
+    {
+        // retry
+    }
+    else
+    {
+        // fail
+    }
+}
+```
+
+##### Provisioning Service Errors
+
+Each Provisioning operation will return a status encoded as `az_iot_provisioning_client_status`.
+After the register operation is complete, the `az_iot_provisioning_client_registration_state registration_information` field will contain the `az_iot_provisioning_client_status error_code` that represents the overall status of the registration operation.  
+
+The following APIs may be used to determine if the status indicates error and if the error should be retried:
+
+```C
+az_iot_provisioning_client_status status = result.registration_information.error_code;
+if (az_iot_provisioning_client_is_success_status(status))
+{
+    // success case 
+}
+else
+{
+    if (az_iot_provisioning_client_is_retriable_status(status))
+    {
+        // retry
+    }
+    else
+    {
+        // fail
+    }
+}
+```
+
+#### Retry timing
+
+Network timeouts and the MQTT keep-alive interval should be configured considering tradeoffs between how fast network issues are detected vs traffic overheads. [This document](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#default-keep-alive-timeout) describes the recommended keep-alive timeouts as well as the minimum idle timeout supported by Azure IoT services.
+
+For connectivity issues at all layers (TCP, TLS, MQTT) as well as cases where there is no `retry-after` sent by the service, we suggest using an exponential back-off with random jitter function. `_az_retry_calc_delay` is available in Azure SDK Core:
+
+```C
+// random_ms is a random value between 0 and max_jitter_msec

```

Comment:
Consider adding " that the application calculates" or even some pseoudo-code that it's clear APP writes.  I get it's bordering on over-communication, but what SDK does and app does is more nuanced in this world.  And especially because we recommend a default for max_jitter_msec below, I worry that a dev could search in vain for our helper to determine this unless we're clear (or unless we move jitter into the API below per your comment to Jeffrey.